messaging/broker.go

```diff
@@ -965,30 +977,34 @@ func (t *Topic) Truncate(maxSize int64) (int, int64, error) {
 			break
 		}
 
-		// More than 2 segments are needed for current writes and replication checks.
-		if len(segments) <= 2 {
-			break
-		}
-
-		// Attempt deletion of oldest segment in topic. First and second segment needed
-		// for this operation.
-		first := segments[0]
-		second := segments[1]
+		// A segment can be deleted under two conditions:
+		//
+		//   1) the segment is too stale
+		//   2) the last entry has been replicated
+		//
+		// The most efficient way to see if the last entry has been replicated
+		// is to check if first entry of next segment has been replicated.
+		if len(segments) > 1 && t.MaxSegmentAge > 0 && time.Since(segments[0].ModTime) > t.MaxSegmentAge {
```

Comment:
Yeah, exactly. We do need to policy like this, or data would grow forever on the brokers. I too want to be sure about introducing it now.